### PR TITLE
[flash_ctrl,dv] more tb cleanup for reset event

### PR DESCRIPTION
**Pull request by jdonjdon**
following tests need scoreboard cleanup during intermittent reset event

### CODE REVIEW

**Review by eshapira — APPROVED** [review]
LGTM, thanks!

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_derr_detect_vseq.sv
Comment (marnovandermaas) [inline hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_derr_detect_vseq.sv:60]: What's the point of wrapping a fork in another fork?

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_prog_reset_vseq.sv
Comment (marnovandermaas) [inline hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_prog_reset_vseq.sv:91]: Can you explain why this was necessary?

**Review by marnovandermaas — COMMENTED** [review]
Some clarification questions from my side.

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_derr_detect_vseq.sv
Comment (jdonjdon) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_derr_detect_vseq.sv:60]: Ah, outer fork define the scope of `disable fork`.
Otherwise disable fork may disable every fork in the sim.

Comment (marnovandermaas) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_derr_detect_vseq.sv:60]: Ah that makes sense.

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_prog_reset_vseq.sv
Comment (jdonjdon) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_prog_reset_vseq.sv:91]: If there is multiple reset events in the sequence, scoreboard has to be initialized
for the next simulation task upon the reset event.
This routine clear some blocking event in the scoreboard.
Most sequence don't need it because there is no multiple reset events or 
test ends after the reset if there is.

Comment (marnovandermaas) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_prog_reset_vseq.sv:92]: Thanks!

**Review by marnovandermaas — APPROVED** [review]
Thanks for the clarification, looks good to me. Note: I have not tested this code and we should wait for CI to pass before merging.

**Pull request merged**
